src/main/java/net/coderbot/iris/uniforms/HardcodedCustomUniforms.java

```diff
@@ -50,6 +55,18 @@ private static SmoothedFloat rainStrengthS(FrameUpdateNotifier updateNotifier) {
 		return new SmoothedFloat(15, CommonUniforms::getRainStrength, updateNotifier);
 	}
 
+	private static float getRawPrecipitation() {
+		Biome.Precipitation precipitation = Minecraft.getInstance().level.getBiome(Minecraft.getInstance().player.blockPosition()).getPrecipitation();
```

Comment:
Might want to use the view entity instead of `Minecraft.getInstance().player`. Also, could getBiome ever return null? It's worth a check.